src/renderer/atlas/AtlasEngine.api.cpp

```diff
@@ -869,3 +868,25 @@ void AtlasEngine::_resolveFontMetrics(const FontInfoDesired& fontInfoDesired, Fo
         fontMetrics->colorGlyphs = fontInfoDesired.GetEnableColorGlyphs();
     }
 }
+
+bool AtlasEngine::_updateWithNearbyFontCollection()
+{
+    // _resolveFontMetrics() checks `_api.s->font->fontCollection` for a pre-existing font collection,
+    // before falling back to using the system font collection. This way we can inject our custom one.
+    // Doing it this way is a bit hacky, but it does have the benefit that we can cache a font collection
+    // instance across font changes, like when zooming the font size rapidly using the scroll wheel.
+    wil::com_ptr<IDWriteFontCollection> collection;
+    try
+    {
+        collection = FontCache::GetCached();
+    }
+    CATCH_LOG();
+
+    if (!collection || _api.s->font->fontCollection == collection)
+    {
+        return false;
+    }
+
+    _api.s.write()->font.write()->fontCollection = std::move(collection);
+    return true;
+}

```

Comment:
This helps us avoid retrying when it's pointless. We only need to retry if the font collection actually changed after all.